inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -2058,41 +1832,136 @@ GAPI_FLUID_KERNEL(FScalePlaneArea8u, ScalePlaneArea8u, true) {
+        pckg.include<typename choose_impl<isa_tag_t>::FMerge2>();
+        pckg.include<typename choose_impl<isa_tag_t>::FMerge3>();
+        pckg.include<typename choose_impl<isa_tag_t>::FMerge4>();
+        //at the moment type_dispatch requires something to be returned by the lambda
+        return true;
+    }
+};
+
+struct Split_ResizeISA {
+    cv::gapi::GKernelPackage& pckg;
+
+    Split_ResizeISA(cv::gapi::GKernelPackage& _pckg) : pckg(_pckg) {}
+
+    template<typename isa_tag_t>
+    inline bool operator()(type_to_type<isa_tag_t>) {
+        pckg.include<typename choose_impl<isa_tag_t>::FSplit2>();
+        pckg.include<typename choose_impl<isa_tag_t>::FSplit3>();
+        pckg.include<typename choose_impl<isa_tag_t>::FSplit4>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlane8u>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlane32f>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlanes>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlanes4>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlaneArea8u>();
+        pckg.include<typename choose_impl<isa_tag_t>::FScalePlaneArea32f>();
+        pckg.include<typename choose_impl<isa_tag_t>::FUpscalePlaneArea8u>();
+        pckg.include<typename choose_impl<isa_tag_t>::FUpscalePlaneArea32f>();
+        //at the moment type_dispatch requires something to be returned by the lambda
+        return true;
+    }
+};
+}  //namespace
+
+inline cv::gapi::GKernelPackage FKernelsChooseISA() {
+    // At the moment AVX512 implementation of wide universal intrinsics is slower than AVX2.
+    // So, disable it for now.
+    using isas = remove_t<isas_set, avx512_tag>;
+
+    cv::gapi::GKernelPackage pckg1, pckg2;
+    CC_and_MergeISA ccISA{ pckg1 };
+    Split_ResizeISA sISA{ pckg2 };
+
+    type_dispatch<isas>(is_isa_present{}, ccISA, false);
+    type_dispatch<isas_set>(is_isa_present{}, sISA, false);
+
+    return combine(pckg1, pckg2);
+}
+
+//----------------------------------------------------------------------
+
+GAPI_COMPOUND_KERNEL(FScalePlane, ScalePlane) {
+    static cv::GMat expand(cv::GMat in, int type, const Size& szIn, const Size& szOut, int interp) {
+        GAPI_DbgAssert(CV_8UC1 == type || CV_32FC1 == type);
+        GAPI_DbgAssert(cv::INTER_AREA == interp || cv::INTER_LINEAR == interp);
+
+        if (cv::INTER_AREA == interp) {
+            bool upscale = szIn.width < szOut.width || szIn.height < szOut.height;
+            if (CV_8UC1 == type) {
+                if (upscale)
+                    return UpscalePlaneArea8u::on(in, szOut, interp);
+                else
+                    return   ScalePlaneArea8u::on(in, szOut, interp);
+            }
+            if (CV_32FC1 == type) {
+                if (upscale)
+                    return UpscalePlaneArea32f::on(in, szOut, interp);
+                else
+                    return   ScalePlaneArea32f::on(in, szOut, interp);
+            }
```

Comment:
Do we need ASSERT here if both conditions are false?

Reply:
Assert already exists at the bottom.

Reply:
ok, i see.